src/otx/core/exporter/diffusion.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Exporter for diffusion models that uses native torch and OpenVINO conversion tools."""
+
+from __future__ import annotations
+
+import logging as log
+from pathlib import Path
+
+import onnx
+import openvino
+import torch
+
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.model.base import OTXModel
+from otx.core.types.export import OTXExportFormatType
+from otx.core.types.precision import OTXPrecisionType
+
+
+class DiffusionOTXModelExporter(OTXNativeModelExporter):
+    """Exporter for diffusion models that uses native torch and OpenVINO conversion tools."""
+
+    def export(  # type: ignore[override]
+        self,
+        model: OTXModel,
+        output_dir: Path,
+        base_model_name: str = "exported_model",
+        export_format: OTXExportFormatType = OTXExportFormatType.OPENVINO,
+        precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        to_exportable_code: bool = False,
+    ) -> dict[str, Path]:
+        """Exports input model to the specified deployable format, such as OpenVINO IR or ONNX.
+
+        Args:
+            model (OTXModel): OTXModel to be exported
+            output_dir (Path): path to the directory to store export artifacts
+            base_model_name (str, optional): exported model name
+            format (OTXExportFormatType): final format of the exported model
+            precision (OTXExportPrecisionType, optional): precision of the exported model's weights
+            to_exportable_code (bool, optional): whether to generate exportable code.
+                Currently not supported by Diffusion task.
+
+        Returns:
+            dict[str, Path]: paths to the exported models
+        """
+        if export_format == OTXExportFormatType.OPENVINO:
+            if to_exportable_code:
+                msg = "Exportable code option is not supported and will be ignored."
+                log.warning(msg)
+            fn = self.to_openvino
+        else:
+            fn = self.to_onnx  # type: ignore[assignment]
+
+        return {"model": fn(model, output_dir, base_model_name, precision)}
```

Comment:
Is it necessary? `export` seems from `OTXModelExporter` in base.py and it just returns outputs not using dict.

Reply:
This class inherits from `OTXNativeModelExporter` which uses `onnx_export_configuration` as an export parameter for ONNX. `export` method in this parent class returns `dict[str, Path]`, so the child class method should have the same return type.

Reply:
No, OTXNativeModelExporter inherits export method from OTXModelExporter and its return type is Path, not dict[str, Path].
https://github.com/openvinotoolkit/training_extensions/blob/0387d8bcd629418a3942e2469c2ba2139560b696/src/otx/core/exporter/base.py#L78-L113

Reply:
Fixed in [457ccd0](https://github.com/openvinotoolkit/training_extensions/pull/3875/commits/457ccd07ef8f422ed039aecea7f25aa0b30befa1)